### ISSUE TITLE
Breaks with new Vite 6

### ISSUE THREAD

**Issue by giacomorebonato** [cyco130/vavite#130]
Looking into this issue

```bash
files generated
  dist/client/sw.js
  dist/client/workbox-4723e66c.js
error during build:
[vite:manifest] Cannot read properties of undefined (reading 'length')
    at Object.generateBundle (file:///Users/grebonato/Code/fastrat/node_modules/.pnpm/vite@6.0.1_@types+node@22.10.0_jiti@2.4.0_terser@5.36.0_tsx@4.19.2_yaml@2.6.1/node_modules/vite/dist/node/chunks/dep-BcnkIxro.js:13576:58)
    at Object.handler (file:///Users/grebonato/Code/fastrat/node_modules/.pnpm/vite@6.0.1_@types+node@22.10.0_jiti@2.4.0_terser@5.36.0_tsx@4.19.2_yaml@2.6.1/node_modules/vite/dist/node/chunks/dep-BcnkIxro.js:50852:15)
    at file:///Users/grebonato/Code/fastrat/node_modules/.pnpm/rollup@4.27.4/node_modules/rollup/dist/es/shared/node-entry.js:20812:40
```

https://github.com/giacomorebonato/fastrat/pull/838

**Comment by cyco130** [comment]
Hi!

I just [upgraded the project to Vite 6](https://github.com/cyco130/vavite/pull/131) and the CI (which is pretty extensive) is green for me. I've also been using Rakkas with Vite 6 since the early beta days with no issues. I'm giving Rakkas as an example because it is in some ways similar to your project (which looks pretty cool, starred!), basically a metaframework that uses parts of vavite.

Are you sure the problem is related to vavite and not to some other component of your project? The attached error doesn't seem to come from vavite, for example.

In other words, I'll need a minimal repro to look into it :)

**Comment by martinkruusement** [comment]
```trace
==> /var/log/api/err1.log <==
      at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
4:24:35 PM [vite] Internal server error: createViteRuntime is not a function
      at file:///var/api/node_modules/@vavite/connect/dist/index.js:66:39
      at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
```
`createViteRuntime` was removed in v6:
 https://github.com/vitejs/vite/discussions/16358#discussioncomment-9872925

**Comment by oldium** [comment]
The original issue from @giacomorebonato is in `vite-plugin-pwa`, see [here](https://github.com/vite-pwa/vite-plugin-pwa/issues/798).

The missing `createViteRuntime` reported by @martinkruusement is a real Vite 6 compatibility issue when `useViteRuntime` is `true`:

https://github.com/cyco130/vavite/blob/8b7d7cc9563c6b3ca3ff5a6f27fabaaeb1495811/packages/reloader/src/index.ts#L86-L92

The experimental API has changed, from this [Pull Request](https://github.com/vitejs/vite/pull/16137) it is best visible what have changed.

I just had a quick look at the repository and it seems that there are no tests for `useViteRuntime=true` @cyco130, so this “breaking” change (not breaking according to Vite devs, because of the experimental nature) went unnoticed.

**Comment by giacomorebonato** [comment]
Thanks @oldium for finding that!
